### PR TITLE
used geometric sum for CartesianCoevaluationMorphismWithGivenRange

### PR DESCRIPTION
**Pull request by mohamed-barakat**

### CODE REVIEW

**Comment by mohamed-barakat** [comment]
FYI: `Algebroids` and `FunctorCategories` pass. One can easily construct examples where the new code is exponentially faster.

File: gap/CompilerLogic.gi [hunk 1]
```diff
@@ -4,6 +4,8 @@
 # Implementations
 #
 
+CapJitAddTypeSignature( "/", [ IsInt, IsInt ], IsInt );
```
Comment (zickgraf) [inline gap/CompilerLogic.gi:7]: This is not true in general. Please either use `QuoInt` or introduce a new function `SafeIntegerDivision` (or similar) to CAP, which asserts that the result of the division is an integer.

File: gap/SkeletalFinSetsForCAP.gi [hunk 2]
```diff
@@ -898,12 +898,18 @@ end );
...
     n := Length( N );
     
-    return MapOfFinSets( cat, M, List( [ 0 .. m - 1 ], i -> Sum( [ 0 .. n - 1 ], j -> ( i + m * j ) * (m*n)^j ) ), HN_MxN );
+    mn := m * n;
+    
+    if mn = 1 then
+        return IdentityMorphism( cat, M );
+    fi;
+    
+    return MapOfFinSets( cat, M, List( [ 0 .. m - 1 ], i -> ( ( ( i - m - 1 + mn ) * mn - i ) * mn^n - ( i - m ) * mn + i ) / ( mn - 1 )^2 ), HN_MxN );
```
Comment (zickgraf) [inline gap/SkeletalFinSetsForCAP.gi:912]: A suggestion: You could outsource this to a function `GeometricSum`, which would make the intent more clear. Then you could decide if you want the compiler to resolve this function or not (keeping it as a primitive hides the case distinction and might allow for more high-level optimizations in the future, but of course also prevents low-level optimizations).

If you don't want to do this, please add a comment saying that this is the geometric sum of ..., so one can understand this code more easily.

File: gap/CompilerLogic.gi (hunk 1 above)
Comment (mohamed-barakat) [reply gap/CompilerLogic.gi]: Deleted.

Comment (zickgraf) [reply gap/CompilerLogic.gi]: > It thought it is safe for this package, but maybe you are thinking of loading this file when compiling packages depending on `SkeletalFinSets`.

Exactly. If you need a hack like this in the future, please put it in the `Precompile*.g` files.

Comment (mohamed-barakat) [reply gap/CompilerLogic.gi]: Yes, good idea.

**Pull request merged**